python_scripts/trees.py

```diff
@@ -185,361 +185,126 @@ def plot_decision_function(X, y, clf, ax=None):
+    f"Chinstrap: {48 / 138:.3f}\n"
+    f"Gentoo: {86 / 138:.3f}\n"
 )
 
-
 # %% [markdown]
-# Now that we are able to quantify any split, we can evaluate all possible
-# splits and compute the information gain for each split.
-
-# %%
-splits_information_gain = []
-possible_thresholds = np.sort(single_feature.unique())[1:-1]
-for threshold_value in possible_thresholds:
-    mask_below_threshold = single_feature < threshold_value
-    labels_below_threshold = y_train.loc[mask_below_threshold]
-    labels_above_threshold = y_train.loc[~mask_below_threshold]
-    splits_information_gain.append(
-        information_gain(labels_below_threshold, labels_above_threshold)
-    )
+# It is also important to note that the culmen depth has been disregarded for
+# the moment. It means that whatever the value given, it will not be used
+# during the prediction.
 
 # %%
-plt.plot(possible_thresholds, splits_information_gain)
-plt.xlabel(single_feature.name)
-_ = plt.ylabel("Information gain")
+tree.predict_proba([[50, 10000]])
 
 # %% [markdown]
-# As previously mentioned, we would like to find the threshold value maximizing
-# the information gain. Below we draw a line in the plot, where the maximum
-# information gain value is.
+# Going back to our classification problem, the split found with a maximum
+# depth of 1 is not powerful enough to separate the three species and the model
+# accuracy is low when compared to the linear model.
 
 # %%
-best_threshold_indice = np.argmax(splits_information_gain)
-best_threshold_value = possible_thresholds[best_threshold_indice]
-
-_, ax = plt.subplots()
-ax.plot(possible_thresholds, splits_information_gain)
-ax.set_xlabel(single_feature.name)
-ax.set_ylabel("Information gain")
-ax.axvline(best_threshold_value, color="tab:orange", linestyle="--")
-ax.set_title(f"Best threshold: {best_threshold_value} mm")
+print(
+    f"Accuracy of the {tree.__class__.__name__}: "
+    f"{tree.fit(X_train, y_train).score(X_test, y_test):.2f}"
+)
 
 # %% [markdown]
-# Using this brute-force search, we find that the threshold maximizing the
-# information gain is 43.3 mm.
-#
-# Let's check if this result is similar to the one found with the
-# `DecisionTreeClassifier` from scikit-learn.
+# Indeed, it is not a surprise. We saw earlier that a single feature will not
+# be able to separate all three species. However, from the previous analysis we
+# saw that by using both features we should be able to get fairly good results.
+# Considering the splitting mechanism of the decision tree illustrated above,
+# we should repeat the partitioning on the resulting rectangles created by the
+# first split. In this regard, we expect that the two partitions at the second
+# level of the tree will be using the feature "Culmen Depth".
 
 # %%
-from sklearn.tree import plot_tree
-
-tree = DecisionTreeClassifier(criterion="entropy", max_depth=1)
-tree.fit(single_feature.to_frame(), y_train)
-_ = plot_tree(tree)
-
+tree.set_params(max_depth=2)
```

Comment:
```suggestion
tree = DecisionTreeClassifier(max_depth=2)
```

seems less magical this way.